diff --git a/manjaro-gnome-settings/PKGBUILD b/manjaro-gnome-settings/PKGBUILD
index ed16f1e..6cc7c6a 100644
--- a/manjaro-gnome-settings/PKGBUILD
+++ b/manjaro-gnome-settings/PKGBUILD
@@ -14,6 +14,7 @@ source=("git+$url.git"
 		'manjaro-gnome-messages.script'
 		'manjaro-gdm-branding-install.hook'
 		'manjaro-gdm-branding-remove.hook'
+                '001-vivaldi.patch'
 )
 makedepends=('git')
 sha256sums=('SKIP'
@@ -21,7 +22,8 @@ sha256sums=('SKIP'
             '588f024527bcc54ecb6d6f5d1c6c4879c400ec37cacabd2547540aadf0e0bda7'
             '133abdada6c6438f6d735f8aceba2b4ccef216892969c58f89445b108af5f266'
             '37c9338b05824d0371f09fa3f19df13902c436367d4bf01d55ebeae4a4966060'
-            '9696a9efdb9c36c99f00eae025405657cbb036cee2b18c7d2039598e8f57e05c')
+            '9696a9efdb9c36c99f00eae025405657cbb036cee2b18c7d2039598e8f57e05c'
+            '75fed49601bf1efe76a5e9dc901339054f86e8ada491ee49f1623f6664f67973')
 
 pkgver() {
 	date +%Y%m%d
@@ -122,3 +124,16 @@ package_manjaro-gdm-branding() {
 
 #	_inst_dir '/minimal'
 #}
+
+prepare() {
+  cd ${srcdir}/${pkgbase}
+  local src
+  for src in "${source[@]}"; do
+      src="${src%%::*}"
+      src="${src##*/}"
+      [[ $src = *.patch ]] || continue
+      msg2 "Applying patch: $src..."
+      patch -Np1 -i ../$src
+  done
+}
+
