diff --git a/PKGBUILD b/PKGBUILD
index 6d0151e..80e2b42 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -45,6 +45,9 @@ build() {
     # any kernel but the current.
     make clean
     make -C /usr/lib/modules/${_kernver}/build M="$srcdir/${_module}" modules
+    SIG_HASH=$(grep -o "CONFIG_MODULE_SIG_HASH.*" /usr/lib/modules/$_kernver/build/include/config/auto.conf)
+    SIG_HASH=${SIG_HASH#CONFIG_MODULE_SIG_HASH=}
+    /usr/lib/modules/${_kernver}/build/scripts/sign-file $SIG_HASH /var/lib/dkms/signing_key-${_kernver}.pem /var/lib/dkms/signing_key-${_kernver}.x509 *.ko
 }
 
 package() {
@@ -52,5 +55,16 @@ package() {
 
     install -Dm644 "blacklist-rtl8xxxu.conf" "${pkgdir}/usr/lib/modprobe.d/${_linuxprefix}-blacklist-rtl8xxxu.conf"
     install -Dm644 "${_module}"/*.ko -t "$pkgdir/usr/lib/modules/${_kernver}/extramodules/"
-    find "${pkgdir}" -name '*.ko' -exec zstd --rm -19 {} +
+    COMPRESS=$(grep -o "CONFIG_MODULE_COMPRESS_.* 1" /usr/lib/modules/$_kernver/build/include/generated/autoconf.h|grep -v ALL)
+    COMPRESS=${COMPRESS#CONFIG_MODULE_COMPRESS_}
+    COMPRESS=${COMPRESS% 1}
+    case $COMPRESS in
+    ZSTD)
+      COMP_CMD="zstd -T0 -22 --ultra --auto-threads=physical --rm -f"
+      ;;
+    XZ)
+      COMP_CMD="xz -T0 -f -v -9 --lzma2=dict=2MiB"
+      ;;
+    esac
+    find "${pkgdir}" -name '*.ko' -exec $COMP_CMD {} +
 }
