diff --git a/PKGBUILD b/PKGBUILD
index 16eec70..b3fde62 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -42,14 +42,28 @@ build() {
   # https://bugs.archlinux.org/task/54975 (kernel has no _GLOBAL_OFFSET_TABLE_):
   # Clear EXTRA_CFLAGS since it defaults to injecting CFLAGS and -fno-plt breaks the modules
   make HDAPS=1 KVER="${_kernver}" EXTRA_CFLAGS=
+  SIG_HASH=$(grep -o "CONFIG_MODULE_SIG_HASH.*" /usr/lib/modules/$_kernver/build/include/config/auto.conf)
+  SIG_HASH=${SIG_HASH#CONFIG_MODULE_SIG_HASH=}
+  /usr/lib/modules/${_kernver}/build/scripts/sign-file $SIG_HASH /var/lib/dkms/signing_key-${_kernver}.pem /var/lib/dkms/signing_key-${_kernver}.x509 ${_module}.ko
 }
 
 package() {
   _kernver="$(cat /usr/src/${_linuxprefix}/version)"
 
   find . -name "*.ko" -exec install -Dt "$pkgdir/usr/lib/modules/${_kernver}/extramodules" {} +
-  find "$pkgdir" -name "*.ko" -exec strip --strip-debug {} +
-  find "$pkgdir" -name "*.ko" -exec xz {} +
+  COMPRESS=$(grep -o "CONFIG_MODULE_COMPRESS_.* 1" /usr/lib/modules/$_kernver/build/include/generated/autoconf.h)
+  COMPRESS=${COMPRESS#CONFIG_MODULE_COMPRESS_}
+  COMPRESS=${COMPRESS% 1}
+  case $COMPRESS in
+  ZSTD)
+    COMP_CMD="zstd -T0 -22 --ultra --auto-threads=physical --rm -f"
+    ;;
+  XZ)
+    COMP_CMD="xz -T0 -f -v -9 --lzma2=dict=2MiB"
+    ;;
+  esac
+  find "${pkgdir}" -name "*.ko" -exec strip --strip-debug {} +
+  find "${pkgdir}" -name '*.ko' -exec $COMP_CMD {} +
 
   # load module on startup
   echo tp_smapi | install -Dm644 /dev/stdin "$pkgdir/usr/lib/modules-load.d/$pkgname.conf"
