diff --git a/PKGBUILD b/PKGBUILD
index eb91fc1..3714784 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -25,6 +25,9 @@ sha256sums=('8b1902a94395c2fa5a97f81c94868a9cbc46a48e12309ad01626439bde96f1d9')
 build() {
   cd "${_pkgname}-${pkgver}"
   make KVER="${_kernver}"
+  SIG_HASH=$(grep -o "CONFIG_MODULE_SIG_HASH.*" /usr/lib/modules/$_kernver/build/include/config/auto.conf)
+  SIG_HASH=${SIG_HASH#CONFIG_MODULE_SIG_HASH=}
+  /usr/lib/modules/${_kernver}/build/scripts/sign-file $SIG_HASH /var/lib/dkms/signing_key-${_kernver}.pem /var/lib/dkms/signing_key-${_kernver}.x509 ${_pkgname}.ko
 }
 
 package() {
@@ -32,7 +35,18 @@ package() {
 
   install -dm 755 "${pkgdir}"/usr/lib/{modules/${_kernver}/extramodules,modules-load.d}
   install -m 644 acpi_call.ko "${pkgdir}"/usr/lib/modules/${_kernver}/extramodules/
-  gzip "${pkgdir}"/usr/lib/modules/${_kernver}/extramodules/acpi_call.ko
+  COMPRESS=$(grep -o "CONFIG_MODULE_COMPRESS_.* 1" /usr/lib/modules/$_kernver/build/include/generated/autoconf.h)
+  COMPRESS=${COMPRESS#CONFIG_MODULE_COMPRESS_}
+  COMPRESS=${COMPRESS% 1}
+  case $COMPRESS in
+  ZSTD)
+    COMP_CMD="zstd -T0 -22 --ultra --auto-threads=physical --rm -f"
+    ;;
+  XZ)
+    COMP_CMD="xz -T0 -f -v -9 --lzma2=dict=2MiB"
+    ;;
+  esac
+  find "${pkgdir}" -name '*.ko' -exec $COMP_CMD {} +
   echo acpi_call > "${pkgdir}"/usr/lib/modules-load.d/${pkgname}.conf
 
   install -dm 755 "${pkgdir}"/usr/share/${pkgname}
