diff --git a/PKGBUILD b/PKGBUILD
index b022c8d..8fa6339 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -36,6 +36,10 @@ build() {
 
     cd "${_pkg}"
     make -C kernel SYSSRC=/usr/lib/modules/"${_kernver}/build" module
+    for file in kernel/*.ko
+    do
+        /usr/lib/modules/${_kernver}/build/scripts/sign-file sha512 /var/lib/dkms/signing_key.pem /var/lib/dkms/signing_key.x509 $file
+    done
 }
 
 package() {
