diff --git a/PKGBUILD b/PKGBUILD
index 1b7ca90..c8c8203 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -45,6 +45,7 @@ build() {
       M="$srcdir/$_pkgname-$pkgver/src" \
       EXTRA_CFLAGS="-DCONFIG_R8168_NAPI -DCONFIG_R8168_VLAN -DCONFIG_ASPM -DENABLE_S5WOL -DENABLE_EEE" \
       modules
+    /usr/lib/modules/${_kernver}/build/scripts/sign-file sha512 /var/lib/dkms/signing_key.pem /var/lib/dkms/signing_key.x509 ${_pkgname}.ko
 }
 
 package() {
