diff --git a/PKGBUILD b/PKGBUILD
index 7ed6961..53df7dd 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -27,6 +27,7 @@ build() {
   _kernver="$(cat /usr/lib/modules/${_extramodules}/version)"
   cd "${_pkgname}-${pkgver}"
   make KVER="${_kernver}"
+  /usr/lib/modules/${_kernver}/build/scripts/sign-file sha512 /var/lib/dkms/signing_key-${_kernver}.pem /var/lib/dkms/signing_key-${_kernver}.x509 ${_pkgname}.ko
 }
 
 package() {

