diff --git a/PKGBUILD b/PKGBUILD
index ad199f9..94bee57 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -30,6 +30,34 @@ build() {
   # build host modules
   echo 'Host modules'
   fakeroot dkms build --dkmstree "$srcdir" -m vboxhost/${pkgver}_OSE -k ${_kernver}
+  cd "$srcdir/vboxhost/$_pkgver/$_kernver/$CARCH/module"
+  for f in *.ko.*
+  do
+          case "$f" in
+          *.zst)
+                zstd -d $f
+                COMPRESS=zstd
+                ;;
+          *.xz)
+                xz -d $f
+                COMPRESS=xz
+                ;;
+          esac
+  done
+
+  for f in *.ko
+  do
+          /usr/lib/modules/${_kernver}/build/scripts/sign-file sha512 /var/lib/dkms/signing_key-${_kernver}.pem /var/lib/dkms/signing_key-${_kernver}.x509 $f
+  done
+
+  case "$COMPRESS" in
+  zstd)
+          zstd -f -22 --ultra -T0 *.ko
+          ;;
+  xz)
+          xz -T0 -9 -e *.ko
+          ;;
+  esac
 }
 
 package(){
@@ -38,8 +66,18 @@ package(){
   cd "vboxhost/${pkgver}_OSE/$_kernver/$CARCH/module"
   install -Dm644 * -t "$pkgdir/usr/lib/modules/$_extramodules/"
 
-  # compress each module individually
-  find "$pkgdir" -name '*.ko' -exec xz -T1 {} +
+  COMPRESS=$(grep -o "CONFIG_MODULE_COMPRESS_.* 1" /usr/lib/modules/$_kernver/build/include/generated/autoconf.h)
+  COMPRESS=${COMPRESS#CONFIG_MODULE_COMPRESS_}
+  COMPRESS=${COMPRESS% 1}
+  case $COMPRESS in
+  ZSTD)
+    COMP_CMD="zstd -T0 -22 --ultra --auto-threads=logical --rm -f"
+    ;;
+  XZ)
+    COMP_CMD="xz -T0 -f -q -9 --lzma2=dict=2MiB"
+    ;;
+  esac
+  find "${pkgdir}" -name '*.ko' -exec $COMP_CMD {} +
 
   # systemd module loading
   printf '%s\n' vboxdrv vboxnetadp vboxnetflt |
