diff --git a/PKGBUILD b/PKGBUILD
index fff61a5..88e7543 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -12,22 +12,58 @@ arch=('x86_64')
 url="http://zfsonlinux.org/"
 license=('CDDL')
 groups=("$_linuxprefix-extramodules")
-depends=("$_linuxprefix" "zfs-utils=${pkgver}")
-makedepends=("$_linuxprefix-headers" "zfs-dkms=${pkgver}")
-provides=("zfs=${pkgver}" "ZFS-MODULE=${pkgver}")
-options=('!strip')
+depends=("$_linuxprefix" "kmod" "zfs-utils=${pkgver}")
+makedepends=("$_linuxprefix-headers")
+provides=("zfs=${pkgver}")
+install=zfs.install
+source=("https://github.com/openzfs/zfs/releases/download/zfs-${pkgver}/zfs-${pkgver}.tar.gz"{,.asc})
+#source=("git+https://github.com/openzfs/zfs.git#commit=${_commit}"
+sha256sums=('76bc0547d9ba31d4b0142e417aaaf9f969072c3cb3c1a5b10c8738f39ed12fc9'
+            'SKIP')
+validpgpkeys=('4F3BA9AB6D1F8D683DC2DFB56AD860EED4598027'  # Tony Hutter (GPG key for signing ZFS releases) <hutter2@llnl.gov>
+              'C33DF142657ED1F7C328A2960AB9E991C6AF658B') # Brian Behlendorf <behlendorf1@llnl.gov>
+
+prepare() {
+    cd "zfs-${pkgver}"
+
+    ./autogen.sh
+    sed -i "s|\$(uname -r)|${_kernver}|g" configure
+}
 
 build() {
     _kernver="$(cat /usr/lib/modules/$_extramodules/version)"
-    fakeroot dkms build --dkmstree "${srcdir}" -m zfs/${pkgver} -k ${_kernver}
+    cd "zfs-${pkgver}"
+    ./configure --prefix=/usr --sysconfdir=/etc --sbindir=/usr/bin --libdir=/usr/lib \
+                --datadir=/usr/share --includedir=/usr/include/ --with-udevdir=/lib/udev \
+                --libexecdir=/usr/lib/zfs-${pkgver} --with-config=kernel \
+                --with-linux=/usr/lib/modules/${_kernver}/build
+    make
 }
 
 package() {
     _kernver="$(cat /usr/lib/modules/$_extramodules/version)"
-    install -Dt "${pkgdir}/usr/lib/modules/${_extramodules}" -m644 zfs/${pkgver}/${_kernver}/${CARCH}/module/*
+    cd "zfs-${pkgver}"
+    install -Dm 644 module/*.ko -t "$pkgdir/usr/lib/modules/$_extramodules/"
+    SIG_HASH=$(grep -o "CONFIG_MODULE_SIG_HASH.*" /usr/lib/modules/$_kernver/build/include/config/auto.conf)
+    SIG_HASH=${SIG_HASH#CONFIG_MODULE_SIG_HASH=}
+    for file in $(find "${pkgdir}" -name '*.ko')
+    do
+        /usr/lib/modules/${_kernver}/build/scripts/sign-file $SIG_HASH /var/lib/dkms/signing_key-${_kernver}.pem /var/lib/dkms/signing_key-${_kernver}.x509 $file
+    done
 
     # compress each module individually
-    find "$pkgdir" -name '*.ko' -exec xz -T1 {} +
+    COMPRESS=$(grep -o "CONFIG_MODULE_COMPRESS_.* 1" /usr/lib/modules/$_kernver/build/include/generated/autoconf.h)
+    COMPRESS=${COMPRESS#CONFIG_MODULE_COMPRESS_}
+    COMPRESS=${COMPRESS% 1}
+    case $COMPRESS in
+    ZSTD)
+        COMP_CMD="zstd -T0 -22 --ultra --auto-threads=physical --rm -f"
+        ;;
+    XZ)
+        COMP_CMD="xz -T0 -f -v -9 --lzma2=dict=2MiB"
+        ;;
+    esac
+    find "${pkgdir}" -name '*.ko' -exec $COMP_CMD {} +
 
     # systemd module loading
     printf '%s\n' spl zfs |
