diff --git a/PKGBUILD b/PKGBUILD
index c326427..d9ef731 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -32,12 +32,27 @@ build() {
 
   cd "$_pkgname-$pkgver"
   make -j1 KDIR=/usr/lib/modules/${_kernver}/build
+  SIG_HASH=$(grep -o "CONFIG_MODULE_SIG_HASH.*" /usr/lib/modules/$_kernver/build/include/config/auto.conf)
+  SIG_HASH=${SIG_HASH#CONFIG_MODULE_SIG_HASH=}
+  /usr/lib/modules/${_kernver}/build/scripts/sign-file $SIG_HASH /var/lib/dkms/signing_key-${_kernver}.pem /var/lib/dkms/signing_key-${_kernver}.x509 vhba.ko
 }
 
 package() {
   cd "$_pkgname-$pkgver"
   install -D vhba.ko "$pkgdir/usr/lib/modules/${_kernver}/extramodules/vhba.ko"
-  find "$pkgdir" -name '*.ko' -exec gzip -9 {} \;
+  COMPRESS=$(grep -o "CONFIG_MODULE_COMPRESS_.* 1" /usr/lib/modules/$_kernver/build/include/generated/autoconf.h)
+  COMPRESS=${COMPRESS#CONFIG_MODULE_COMPRESS_}
+  COMPRESS=${COMPRESS% 1}
+  case $COMPRESS in
+  ZSTD)
+    COMP_CMD="zstd -T0 -22 --ultra --auto-threads=physical --rm -f"
+    ;;
+  XZ)
+    COMP_CMD="xz -T0 -f -v -9 --lzma2=dict=2MiB"
+    ;;
+  esac
+ 
+  find "${pkgdir}" -name '*.ko' -exec $COMP_CMD {} +
   install -Dm 644 "../60-vhba.rules" \
     "$pkgdir/usr/lib/udev/rules.d/60-$_linuxprefix-vhba.rules"
 }
