diff --git a/PKGBUILD b/PKGBUILD
index 45a9396..23c62c3 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -30,6 +30,13 @@ build() {
   # build host modules
   echo 'Host modules'
   fakeroot dkms build --dkmstree "$srcdir" -m vboxhost/${pkgver}_OSE -k ${_kernver}
+  cd "$srcdir/vboxhost/$_pkgver/$_kernver/$CARCH/module"
+  xz -d *.xz
+  for f in *.ko
+  do
+      /usr/lib/modules/${_kernver}/build/scripts/sign-file sha512 /var/lib/dkms/signing_key-${_kernver}.pem /var/lib/dkms/signing_key-${_kernver}.x509 $f
+  done
+  xz -9 -e -T0 *.ko
 }
 
 package(){
