diff --git a/PKGBUILD b/PKGBUILD
index edf184a..3e6ce6e 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -34,6 +34,7 @@ build() {
   cd ${_pkgname}-${pkgver}
   # KDIR is necessary even when cleaning
   make KDIR=/usr/lib/modules/${_kernver}/build
+  /usr/lib/modules/${_kernver}/build/scripts/sign-file sha512 /var/lib/dkms/signing_key-${_kernver}.pem /var/lib/dkms/signing_key-${_kernver}.x509 ${_pkgname}.ko
 }
 
 package() {
