diff --git a/PKGBUILD b/PKGBUILD
index d9b2eaf..e735b3a 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -52,6 +55,10 @@ build() {
 
     cd "${_pkg}"
     make -C kernel SYSSRC=/usr/lib/modules/"${_kernver}/build" module
+    for file in kernel/*.ko
+    do
+        /usr/lib/modules/${_kernver}/build/scripts/sign-file sha512 /var/lib/dkms/signing_key-${_kernver}.pem /var/lib/dkms/signing_key-${kernver}.x509 $file
+    done
 }
 
 package() {
