diff --git a/PKGBUILD b/PKGBUILD
index 6658a1a..759bede 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -50,6 +50,7 @@ build() {
   # https://bugs.archlinux.org/task/54975 (kernel has no _GLOBAL_OFFSET_TABLE_):
   # Clear EXTRA_CFLAGS since it defaults to injecting CFLAGS and -fno-plt breaks the modules
   make HDAPS=1 KVER="$(</usr/lib/modules/${_extraver}/version)" EXTRA_CFLAGS=
+  /usr/lib/modules/${_kernver}/build/scripts/sign-file sha512 /var/lib/dkms/signing_key-${_kernver}.pem /var/lib/dkms/signing_key-${_kernver}.x509 ${_pkgname}.ko
 }
 
 package() {
@@ -57,8 +58,19 @@ package() {
   find . -name "*.ko" -exec install -Dt "$pkgdir$_extramodules" {} +
 
   # compress kernel modules
-  find "$pkgdir" -name "*.ko" -exec strip {} +
-  find "$pkgdir" -name "*.ko" -exec xz {} +
+  COMPRESS=$(grep -o "CONFIG_MODULE_COMPRESS_.* 1" /usr/lib/modules/$_kernver/build/include/generated/autoconf.h)
+  COMPRESS=${COMPRESS#CONFIG_MODULE_COMPRESS_}
+  COMPRESS=${COMPRESS% 1}
+  case $COMPRESS in
+  ZSTD)
+    COMP_CMD="zstd -T0 -22 --ultra --auto-threads=physical --rm -f"
+    ;;
+  XZ)
+    COMP_CMD="xz -T0 -f -v -9 --lzma2=dict=2MiB"
+    ;;
+  esac
+  find "${pkgdir}" -name "*.ko" -exec strip {} +
+  find "${pkgdir}" -name '*.ko' -exec $COMP_CMD {} +
 
   # load module on startup
   echo tp_smapi > "../${_pkgname}.conf"
