diff --git a/PKGBUILD b/PKGBUILD
index 9872ae0..1f20e89 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -37,6 +37,16 @@ build() {
   msg2 "$CFLAGS, ${CXXFLAGS}"
 
   fakeroot dkms build --dkmstree "$srcdir" -m "broadcom-wl/$pkgver" -k "$_kernver"
+  cd "$srcdir/$_pkgname/$pkgver/$_kernver/$CARCH/module"
+  if [ -f *.zst ] ;then
+          zstd -d *.zst
+          /usr/lib/modules/${_kernver}/build/scripts/sign-file sha512 /var/lib/dkms/signing_key-${_kernver}.pem /var/lib/dkms/signing_key-${_kernver}.x509 *.ko
+          zstd -f -22 --ultra -T0 *.ko
+  elif [ -f *.xz ] ;then
+          xz -d *.xz
+          /usr/lib/modules/${_kernver}/build/scripts/sign-file sha512 /var/lib/dkms/signing_key-${_kernver}.pem /var/lib/dkms/signing_key-${_kernver}.x509 *.ko
+          xz -T0 -9 -e *.ko
+  fi
 }
 
 package(){
