diff --git a/PKGBUILD b/PKGBUILD
index 3b6e25a..d411ba8 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -5,15 +5,16 @@
 # Contributor: Drew DeVault
 
 _pkgbase=nginx
+_commit=f4279edda9fd
 pkgbase=nginx-quic
 pkgname=(nginx-quic nginx-quic-src)
-pkgver=1.23.3
-pkgrel=1
+pkgver=1.23.4
+pkgrel=7
 pkgdesc='Lightweight HTTP server and IMAP/POP3 proxy server, HTTP/3 QUIC branch'
 arch=('i686' 'x86_64')
 url='https://nginx.org'
 license=('custom')
-depends=('geoip' 'libxcrypt' 'mailcap' 'pcre2' 'zlib')
+depends=('geoip' 'libxcrypt' 'pcre2')
 makedepends=('cmake' 'git' 'go' 'mercurial')
 backup=('etc/nginx/fastcgi.conf'
         'etc/nginx/fastcgi_params'
@@ -27,12 +28,12 @@ backup=('etc/nginx/fastcgi.conf'
 install=nginx.install
 provides=('nginx' 'nginx-mainline')
 conflicts=('nginx')
-source=("hg+https://hg.nginx.org/nginx-quic#revision=91ad1abfb285"
-        "git+https://boringssl.googlesource.com/boringssl#commit=1740ff90a7da10408e6175b516946b392ec8f11f"
+source=("hg+https://hg.nginx.org/nginx-quic#revision=$_commit&#branch=quic"
+        "https://cdn.openbsd.org/pub/OpenBSD/LibreSSL/libressl-3.7.0.tar.gz"
         "service"
         "logrotate")
 sha256sums=('SKIP'
-            'SKIP'
+            '3fc1290f4007ec75f6e9acecbb25512630d1b9ab8c53ba79844e395868c3e006'
             '05fdc0c0483410944b988d7f4beabb00bec4a44a41bd13ebc9b78585da7d3f9b'
             'b9af19a75bbeb1434bba66dd1a11295057b387a2cbff4ddf46253133909c311e')
 
@@ -83,15 +84,15 @@ prepare() {
 
 build() {
   # Clear -D_FORTIFY_SOURCE from C++ build flags, it causes Boringssl tests to fail to compile
-  export CPPFLAGS=${CPPFLAGS/-D_FORTIFY_SOURCE=[1-9]/-D_FORTIFY_SOURCE=0}
-  export CXXFLAGS=${CXXFLAGS/-D_FORTIFY_SOURCE=[1-9]/-D_FORTIFY_SOURCE=0}
+  #export CPPFLAGS=${CPPFLAGS/-D_FORTIFY_SOURCE=[1-9]/-D_FORTIFY_SOURCE=0}
+  #export CXXFLAGS=${CXXFLAGS/-D_FORTIFY_SOURCE=[1-9]/-D_FORTIFY_SOURCE=0}
 
   export CXXFLAGS="$CXXFLAGS -fPIC"
   export CFLAGS="$CFLAGS -fPIC"
 
   if [[ $CC == "clang" ]];then
     _cc_opt="-flto"
-    _ld_opt="-flto -fuse-ld=lld"
+    _ld_opt="$LDFLAGS -flto -fuse-ld=lld"
   else
     _cc_opt=""
     _ld_opt=""
@@ -100,12 +101,9 @@ build() {
     # -Wno-array-bounds: 2022-05-21 for compatiblity with GCC 12.1 (https://bugs.chromium.org/p/boringssl/issues/detail?id=492&sort=-modified)
     export CFLAGS="$CFLAGS -Wno-stringop-overflow -Wno-array-parameter -Wno-array-bounds"
   fi
-
-  cd ${srcdir}/boringssl
-  mkdir build && cd build && cmake -DCMAKE_BUILD_TYPE=Release ../ && make crypto ssl
-  cd ${srcdir}/boringssl
-  mkdir -p .openssl/lib && cd .openssl && ln -s ../include . && cd ../
-  cp ${srcdir}/boringssl/build/crypto/libcrypto.a ${srcdir}/boringssl/build/ssl/libssl.a .openssl/lib && cd ..
+  
+  _cc_opt+=" -I../libressl-3.7.0/include"
+  _ld_opt+=" -L../libressl-3.7.0/build/ssl -L../libressl-3.7.0/build/crypto"
 
   cd ${srcdir}/$pkgname
   ./auto/configure \
@@ -123,14 +121,15 @@ build() {
     --http-fastcgi-temp-path=/var/lib/nginx/fastcgi \
     --http-scgi-temp-path=/var/lib/nginx/scgi \
     --http-uwsgi-temp-path=/var/lib/nginx/uwsgi \
-    --with-openssl=${srcdir}/boringssl \
-    --with-cc-opt="${_cc_opt} -I../boringssl/include" \
-    --with-ld-opt="${_ld_opt} -L../boringssl/build/ssl -L../boringssl/build/crypto" \
+    --with-openssl=${srcdir}/libressl-3.7.0 \
+    --with-cc-opt="${_cc_opt}" \
+    --with-ld-opt="${_ld_opt}" \
     ${_common_flags[@]} \
     ${_mainline_flags[@]} \
     ${_quic_flags[@]}
 
-  touch ${srcdir}/boringssl/.openssl/include/openssl/ssl.h
+    #--with-openssl=${srcdir}/boringssl \
+  #touch ${srcdir}/boringssl/.openssl/include/openssl/ssl.h
   make
 }
 
@@ -177,7 +176,7 @@ package_nginx-quic-src() {
   arch=('any')
   provides=('nginx-src' 'nginx-mainline-src')
   conflicts=($_pkgbase-src)
-  depends=()
+  depends=('perl')
   backup=()
   install -d "$pkgdir/usr/src"
   test -d "$pkgdir/usr/src/nginx" && rm -r "$pkgdir/usr/src/nginx"
@@ -187,6 +186,9 @@ package_nginx-quic-src() {
   # Link the 'configure' script to its location in release tarballs,
   # as this is where modules expect it
   ln -s /usr/src/nginx/auto/configure "$pkgdir/usr/src/nginx"
+  cd $pkgname
+  install -d "$pkgdir"/usr/share/licenses/$pkgname
+  ln -s /usr/src/nginx/docs/text/LICENSE "$pkgdir"/usr/share/licenses/$pkgname/LICENSE
 }
 
 # vim:set ts=2 sw=2 et:
