diff --git a/PKGBUILD b/PKGBUILD
index c1e0f98..f4ec2a3 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -20,7 +20,7 @@ makedepends=("broadcom-wl-dkms>=$pkgver"
 groups=("$_linuxprefix-extramodules")
 provides=("$_pkgname=$pkgver")
 install=$_pkgname.install
-backup=('etc/modprobe.d/$_linuxprefix-broadcom-wl.conf')
+backup=("etc/modprobe.d/${_linuxprefix}-broadcom-wl.conf")
 source=(broadcom-wl-dkms.conf)
 sha256sums=('b97bc588420d1542f73279e71975ccb5d81d75e534e7b5717e01d6e6adf6a283')
 
@@ -47,4 +47,5 @@ package(){
   find "$pkgdir" -name '*.ko' -exec gzip -9 {} +
   sed -i -e "s/EXTRAMODULES='.*'/EXTRAMODULES='$_extramodules'/" "$startdir/$_pkgname.install"
   install -D -m 644 "${srcdir}/broadcom-wl-dkms.conf" "${pkgdir}/etc/modprobe.d/${_linuxprefix}-broadcom-wl.conf"
+  install -D -m 644 "/usr/share/licenses/broadcom-wl-dkms/LICENSE" "${pkgdir}/usr/share/licenses/linux64-broadcom-wl/LICENSE"
 }
