diff --git a/PKGBUILD b/PKGBUILD
index ee205aa..f589e59 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -28,8 +28,16 @@ makedepends=('git' "$_linuxprefix-headers")
 provides=("${_pkgname}=$pkgver")
 install="${_pkgname}.install"
 _commit=6e80bb1752280bcd142d86ecd0739661bd0e8312  # tags/tp-smapi/0.44
-source=("git+https://github.com/evgeni/tp_smapi#commit=$_commit")
-sha256sums=('SKIP')
+source=("git+https://github.com/evgeni/tp_smapi#commit=$_commit"
+        'tp_smapi-clang.patch')
+sha256sums=('SKIP'
+            '1be3a832360eab9f3bd7c7aec86b019d951271923fe2ff08de347a6dedea941d')
+
+prepare() {
+  cd $_pkgname
+
+  patch -p1 -i ../tp_smapi-clang.patch
+}
 
 pkgver() {
   cd $_pkgname
diff --git a/tp_smapi-clang.patch b/tp_smapi-clang.patch
new file mode 100644
index 0000000..a2e5325
--- /dev/null
+++ b/tp_smapi-clang.patch
@@ -0,0 +1,13 @@
+diff --git a/tp_smapi.c b/tp_smapi.c
+index de7391e..5be638b 100644
+--- a/tp_smapi.c
++++ b/tp_smapi.c
+@@ -201,7 +201,7 @@ static int smapi_request(u32 inEBX, u32 inECX,
+ 			 "=m"(tmpEDI),
+ 			 "=m"(tmpESI)
+ 			:"m"(inEBX), "m"(inECX), "m"(inEDI), "m"(inESI),
+-			 "m"((u16)smapi_port)
++			 "m"(smapi_port)
+ 			:"%eax", "%ebx", "%ecx", "%edx", "%edi",
+ 			 "%esi");
+ 
