diff --git a/PKGBUILD b/PKGBUILD
index 0f06a4b..14c19a9 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -19,7 +19,7 @@ makedepends=("broadcom-wl-dkms>=$pkgver"
              "$_linuxprefix" "$_linuxprefix-headers")
 groups=("$_linuxprefix-extramodules")
 provides=("$_pkgname=$pkgver")
-backup=('etc/modprobe.d/$_linuxprefix-broadcom-wl.conf')
+backup=("etc/modprobe.d/$_linuxprefix-broadcom-wl.conf")
 source=(broadcom-wl-dkms.conf)
 sha256sums=('b97bc588420d1542f73279e71975ccb5d81d75e534e7b5717e01d6e6adf6a283')
 
