diff --git a/PKGBUILD b/PKGBUILD
index 981061e..8ea750d 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -10,7 +10,7 @@ _linuxprefix=linux67
 
 _module=vhba-module
 pkgname="${_linuxprefix}-${_module}"
-pkgver=20211218
+pkgver=20240202
 pkgrel=17
 pkgdesc="Kernel module that emulates SCSI devices"
 arch=('x86_64')
@@ -21,7 +21,7 @@ makedepends=("${_linuxprefix}-headers")
 provides=("${_module}=$pkgver" "VHBA-MODULE")
 groups=("${_linuxprefix}-extramodules")
 source=("http://downloads.sourceforge.net/cdemu/${_module}-$pkgver.tar.xz")
-sha256sums=('72c5a8c1c452805e4cef8cafefcecc2d25ce197ae4c67383082802e5adcd77b6')
+sha256sums=('bf5850d4b8f50221ca87d7343a929eda87b191f6f5ae8c614174543b5badde83')
 
 build() {
   _kernver="$(cat /usr/src/${_linuxprefix}/version)"
