diff --git a/PKGBUILD b/PKGBUILD
index d104a6c..f053b6b 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -27,8 +27,10 @@ makedepends=('git' "$_linuxprefix-headers")
 groups=("$_linuxprefix-extramodules")
 install="${_pkgname}.install"
 _commit=0c3398b1acf2a2cabd9cee91dc3fe3d35805fa8b  # tags/tp-smapi/0.43
-source=("git+https://github.com/evgeni/tp_smapi#commit=$_commit")
-sha256sums=('SKIP')
+source=("git+https://github.com/evgeni/tp_smapi#commit=$_commit"
+        'tp_smapi-clang.patch')
+sha256sums=('SKIP'
+            'cb5d682450af8ac8436e1645d4fdaa32382ae83cc03c03b0c2f878f78e017be5')
 
 #pkgver() {
 #  cd $_pkgname
@@ -38,6 +40,7 @@ sha256sums=('SKIP')
 prepare() {
   cd $_pkgname
 
+  patch -p1 -i ../tp_smapi-clang.patch
 }
 
 build() {
