diff --git a/PKGBUILD b/PKGBUILD
index 89658c7..951601d 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -27,8 +27,16 @@ makedepends=('git' "$_linuxprefix-headers")
 groups=("$_linuxprefix-extramodules")
 install="${_pkgname}.install"
 _commit=a63729ab30d85430048f65c37f29188ab484cd52  # tags/tp-smapi/0.43
-source=("git+https://github.com/evgeni/tp_smapi#commit=$_commit")
-sha256sums=('SKIP')
+source=("git+https://github.com/evgeni/tp_smapi#commit=$_commit"
+        'tp_smapi-clang.patch')
+sha256sums=('SKIP'
+            'c394a59bb8cd8fb76646ccc3698568759df844ae28b35c07b9e2cd3cf95c739f')
+
+prepare() {
+  cd $_pkgname
+
+  patch -p1 -i ../tp_smapi-clang.patch
+}
 
 pkgver() {
   cd $_pkgname
diff --git a/tp_smapi-clang.patch b/tp_smapi-clang.patch
new file mode 100644
index 0000000..1ba5e58
--- /dev/null
+++ b/tp_smapi-clang.patch
@@ -0,0 +1,13 @@
+diff --git a/tp_smapi.c b/tp_smapi.c
+index 6346287..1e54da2 100644
+--- a/tp_smapi.c
++++ b/tp_smapi.c
+@@ -199,7 +199,7 @@ static int smapi_request(u32 inEBX, u32 inECX,
+ 			 "=m"(tmpEDI),
+ 			 "=m"(tmpESI)
+ 			:"m"(inEBX), "m"(inECX), "m"(inEDI), "m"(inESI),
+-			 "m"((u16)smapi_port)
++			 "m"(smapi_port)
+ 			:"%eax", "%ebx", "%ecx", "%edx", "%edi",
+ 			 "%esi");
+ 
