diff --git a/PKGBUILD b/PKGBUILD
index faec23e..6ab3e7a 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -110,14 +110,15 @@ prepare() {
   for src in "${source[@]}"; do
       src="${src%%::*}"
       src="${src##*/}"
-      [[ $src = *.patch ]] || continue
-      msg2 "Applying patch: $src..."
-      patch -Np1 < "../$src"
+      if [[ $src =~ .*\.patch ]]; then
+              msg2 "Applying patch: $src..."
+              patch -Np1 < "../$src"
+      elif [[ $src =~ .*\.gitpatch ]]; then
+              msg2 "Applying gitpatch: $src..."
+              git --git-dir= apply -p1 --stat --apply "../$src"
+      fi
   done
 
-  msg2 "0513-bootsplash"
-  git apply -p1 < "../0413-bootsplash.gitpatch"
-
   msg2 "add config"
   cat "../config" > ./.config
 
@@ -136,6 +137,8 @@ prepare() {
   msg "get kernel version"
   make prepare
 
+  make menuconfig
+
   msg "rewrite configuration"
   yes "" | make config >/dev/null
 }
