diff --git a/0501-pahole-skip_encoding_btf_enum64.patch b/0501-pahole-skip_encoding_btf_enum64.patch
new file mode 100644
index 0000000..ce91383
--- /dev/null
+++ b/0501-pahole-skip_encoding_btf_enum64.patch
@@ -0,0 +1,13 @@
+diff --git a/scripts/pahole-flags.sh b/scripts/pahole-flags.sh
+index 0d99ef17e4a5..d4f3d63cb434 100755
+--- a/scripts/pahole-flags.sh
++++ b/scripts/pahole-flags.sh
+@@ -20,4 +20,8 @@  if [ "${pahole_ver}" -ge "122" ]; then
+ 	extra_paholeopt="${extra_paholeopt} -j"
+ fi
+ 
++if [ "${pahole_ver}" -ge "124" ]; then
++	extra_paholeopt="${extra_paholeopt} --skip_encoding_btf_enum64"
++fi
++
+ echo ${extra_paholeopt}
diff --git a/PKGBUILD b/PKGBUILD
index 6a726e3..8a79f0c 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -43,7 +43,8 @@ source=("https://www.kernel.org/pub/linux/kernel/v5.x/linux-${_basekernel}.tar.x
         '0410-bootsplash.patch'
         '0411-bootsplash.patch'
         '0412-bootsplash.patch'
-        '0413-bootsplash.gitpatch')
+        '0413-bootsplash.gitpatch'
+        '0501-pahole-skip_encoding_btf_enum64.patch')
 
 sha256sums=('ff240c579b9ee1affc318917de07394fc1c3bb49dac25ec1287370c2e15005a8'
             '1a0cb578fae91e5f25e5c4af0b891b929816cdcee6647f6ffa2d4d9a8ffe4578'
@@ -68,7 +69,8 @@ sha256sums=('ff240c579b9ee1affc318917de07394fc1c3bb49dac25ec1287370c2e15005a8'
             'cf06d959a53eff6d3c287327f1cb2a68346d725cfd1370bc7482a0edc75692fc'
             '27471eee564ca3149dd271b0817719b5565a9594dc4d884fe3dc51a5f03832bc'
             'b6e695edbe349505a89c98054a54443acd90830a312cd035393c5c0a624e45c0'
-            '035ea4b2a7621054f4560471f45336b981538a40172d8f17285910d4e0e0b3ef')
+            '035ea4b2a7621054f4560471f45336b981538a40172d8f17285910d4e0e0b3ef'
+            '9dcafece9e0986d32b7961f3a0b86844ea271e249e9bd868d72d0fa1a019d79e')
 
 prepare() {
   cd "linux-${_basekernel}"
