diff --git a/PKGBUILD b/PKGBUILD
index bbde3c4..2f45677 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -17,7 +17,7 @@ license=('GPL2')
 makedepends=(bc docbook-xsl libelf pahole python-sphinx git inetutils kmod xmlto cpio perl tar xz)
 options=('!strip')
 source=("https://git.kernel.org/torvalds/t/linux-${_basekernel}.tar.gz"
-        "https://www.kernel.org/pub/linux/kernel/v6.x/patch-${pkgver}.xz"
+        "https://cdn.kernel.org/pub/linux/kernel/v6.x/patch-${pkgver}.xz"
         'config'
         # ARCH Patches
         '0101-ZEN_Add_sysctl_and_CONFIG_to_disallow_unprivileged_CLONE_NEWUSER.patch'
@@ -69,11 +69,15 @@ sha256sums=('6246ee76209fb1ff46ffcd67d0cc8029dec2ef929de32ef5460a7a5649583103'
             '27471eee564ca3149dd271b0817719b5565a9594dc4d884fe3dc51a5f03832bc'
             'b6e695edbe349505a89c98054a54443acd90830a312cd035393c5c0a624e45c0'
             '035ea4b2a7621054f4560471f45336b981538a40172d8f17285910d4e0e0b3ef'
-            '5f2d2c4ab326830df11d5bc985feab2988ee929bbf249f85453536e1998951bd')
+            '60fc3f22d49c6aedf614af07582328ad0119166484e6d5042bcd3b40e7745479')
+
+validpgpkeys=(647F28654894E3BD457199BE38DBBDC86092693E)
 
 prepare() {
   cd "linux-${_basekernel}"
 
+  unset RUSTC
+  cp ~/.cargo/bin/bindgen{-0.62.0,}
   # add upstream patch
   msg "add upstream patch"
   patch -p1 -i "../patch-${pkgver}"
@@ -82,14 +86,15 @@ prepare() {
   for src in "${source[@]}"; do
       src="${src%%::*}"
       src="${src##*/}"
-      [[ $src = *.patch ]] || continue
-      msg2 "Applying patch: $src..."
-      patch -Np1 < "../$src"
+      if [[ $src =~ .*\.patch ]]; then
+              msg2 "Applying patch: $src..."
+              patch -Np1 < "../$src"
+      elif [[ $src =~ .*\.gitpatch ]]; then
+              msg2 "Applying gitpatch: $src..."
+              git --git-dir= apply -p1 --stat --ignore-whitespace --apply "../$src"
+      fi
   done
 
-  msg2 "Applying 0999-acs.gitpatch"
-  patch --ignore-whitespace --fuzz 3 -p1 < "../0999-acs.gitpatch"
-
   msg2 "add config"
   cat "../config" > ./.config
 
@@ -108,6 +113,8 @@ prepare() {
   msg "get kernel version"
   make prepare
 
+  make menuconfig
+
   msg "rewrite configuration"
   yes "" | make config >/dev/null
 }
@@ -115,8 +122,10 @@ prepare() {
 build() {
   cd "linux-${_basekernel}"
 
+  unset RUSTC
+  sudo choom -n -1000 -p $$
   msg "build"
-  make ${MAKEFLAGS} LOCALVERSION= bzImage modules
+  make LLVM=1 ${MAKEFLAGS} LOCALVERSION= bzImage modules
 }
 
 package_linux61() {
@@ -153,7 +162,7 @@ package_linux61() {
 
 package_linux61-headers() {
   pkgdesc="Header files and scripts for building modules for ${pkgbase/linux/Linux} kernel"
-  depends=('gawk' 'python' 'libelf' 'pahole')
+  depends=('gawk' 'python' 'libelf' 'pahole' 'python-ply' 'perl')
   provides=("linux-headers=$pkgver")
 
   cd "linux-${_basekernel}"
