--- kernel-open/nvidia/nv.c	2024-12-10 12:23:21.996789108 +0700
+++ kernel-open/nvidia/nv.c	2024-12-10 12:24:01.216789480 +0700
@@ -127,7 +127,7 @@ MODULE_ALIAS_CHARDEV_MAJOR(NV_MAJOR_DEVI
  * DMA_BUF namespace is added by commit id 16b0314aa746
  * ("dma-buf: move dma-buf symbols into the DMA_BUF module namespace") in 5.16
  */
-MODULE_IMPORT_NS(DMA_BUF);
+MODULE_IMPORT_NS("DMA_BUF");
 #endif  // defined(MODULE_IMPORT_NS)
 
 const NvBool nv_is_rm_firmware_supported_os = NV_TRUE;
