diff --git a/mkinitcpio b/mkinitcpio
index d819e6d..3af22ea 100755
--- a/mkinitcpio
+++ b/mkinitcpio
@@ -691,7 +691,7 @@ if [[ $_optgenimg ]]; then
         die 'Unable to write to %s' "$_optgenimg"
     fi
 
-    _optcompress=${_optcompress:-${COMPRESSION:-gzip}}
+    _optcompress=${_optcompress:-${COMPRESSION:-xz}}
     if ! type -P "$_optcompress" >/dev/null; then
         warning "Unable to locate compression method: %s" "$_optcompress"
         _optcompress=cat
diff --git a/man/mkinitcpio.conf.5.txt b/man/mkinitcpio.conf.5.txt
index d206766..987cfaa 100644
--- a/man/mkinitcpio.conf.5.txt
+++ b/man/mkinitcpio.conf.5.txt
@@ -56,7 +56,7 @@ Variables
 	the kernel understands the compression formats yielded by the *gzip*, *bzip2*,
 	*lz4*, *lzop*, *lzma*, and *xz* compressors. As of linux 5.9 the kernel also
 	understands the compression formats yielded by the *zstd* compressor.
-	If unspecified, this setting defaults to *gzip* compression.
+	If unspecified, this setting defaults to *xz* compression.
 	In order to create an uncompressed image, define
 	this variable as *cat*.
 +
diff --git a/mkinitcpio.conf b/mkinitcpio.conf
index 6ea97fd..a27c352 100644
--- a/mkinitcpio.conf
+++ b/mkinitcpio.conf
@@ -52,12 +52,12 @@ FILES=()
 HOOKS=(base udev autodetect modconf block filesystems keyboard fsck)
 
 # COMPRESSION
-# Use this to compress the initramfs image. By default, gzip compression
+# Use this to compress the initramfs image. By default, xz compression
 # is used. Use 'cat' to create an uncompressed image.
 #COMPRESSION="gzip"
 #COMPRESSION="bzip2"
 #COMPRESSION="lzma"
-#COMPRESSION="xz"
+COMPRESSION="xz"
 #COMPRESSION="lzop"
 #COMPRESSION="lz4"
 #COMPRESSION="zstd"
@@ -65,9 +65,10 @@ HOOKS=(base udev autodetect modconf block filesystems keyboard fsck)
 # COMPRESSION_OPTIONS
 # Additional options for the compressor
 #COMPRESSION_OPTIONS=()
+COMPRESSION_OPTIONS=('-e' '-9' '-T0')
 
 # MODULES_DECOMPRESS
 # Decompress kernel modules during initramfs creation.
 # Enable to speedup boot process, disable to save RAM
 # during early userspace. Switch (yes/no).
-#MODULES_DECOMPRESS="yes"
+MODULES_DECOMPRESS="yes"
