diff --unified --recursive --text a/libalpm/scripts/mkinitcpio b/libalpm/scripts/mkinitcpio
--- a/libalpm/scripts/mkinitcpio	2023-03-19 08:24:59.000000000 -0600
+++ b/libalpm/scripts/mkinitcpio	2023-03-23 10:25:32.608441928 -0600
@@ -8,7 +8,7 @@
 
 process_preset() {
     if [[ -n "$pkgbase" && -e "$preset" ]]; then
-        if ! cmp "$preset" > /dev/null 2>&1 <(sed "s|%PKGBASE%|${pkgbase}|g" /usr/share/mkinitcpio/hook.preset); then
+        if ! cmp "$preset" > /dev/null 2>&1 <(sed "s|%KERNELBASE%|${kernelbase}|g" /usr/share/mkinitcpio/hook.preset); then
             if [[ ! -e "$preset.pacsave" ]]; then
                 # save the preset as pacsave
                 mv -- "$preset" "$preset.pacsave" && return 0
@@ -76,7 +76,7 @@
             mv -- "${preset}.pacsave" "$preset"
         else
             # create the preset from the template
-            sed "s|%PKGBASE%|${pkgbase}|g" /usr/share/mkinitcpio/hook.preset \
+            sed "s|%KERNELBASE%|${kernelbase}|g" /usr/share/mkinitcpio/hook.preset \
                 | install -Dm644 /dev/stdin "$preset"
         fi
     fi
@@ -130,6 +130,11 @@
         continue
     fi
 
+    if ! read -r kernelbase > /dev/null 2>&1 < "${line%/vmlinuz}/kernelbase"; then
+        # this kernel has no kernelbase, use pkgbase
+        kernelbase="${pkgbase}"
+    fi
+
     case "$1" in
         install) install_kernel "$pkgbase";;
         remove) remove_kernel "$pkgbase";;
diff --unified --recursive --text a/mkinitcpio.d/example.preset b/mkinitcpio.d/example.preset
--- a/mkinitcpio.d/example.preset	2023-03-19 08:24:59.000000000 -0600
+++ b/mkinitcpio.d/example.preset	2023-03-23 10:29:50.223969082 -0600
@@ -15,14 +15,14 @@
 # presetname_image   - the filename of generated image
 # presetname_options - any extra options
 
-#default_kver="3.0-ARCH"
+#default_kver="5.3.8-3-MANJARO"
 #default_config="/etc/mkinitcpio.conf"
-default_image="/tmp/initramfs-linux.img"
-default_uki="/efi/EFI/Linux/arch-linux.efi"
+default_image="/boot/initramfs-5.3-x86_64.img"
+default_uki="/efi/EFI/Manjaro/grubx64.efi"
 default_options=""
 
-#fallback_kver="3.0-ARCH"
+#fallback_kver="5.3.8-3-MANJARO
 #fallback_config="/etc/mkinitcpio.conf"
-fallback_image="/tmp/initramfs-linux-fallback.img"
-fallback_uki="/efi/EFI/Linux/arch-linux-fallback.efi"
+fallback_image="boot/initramfs-5.3-x86_64-fallback.img"
+fallback_uki="/efi/EFI/Manjaro/grubx64-fallback.efi"
 fallback_options="-S autodetect"
diff --unified --recursive --text a/mkinitcpio.d/hook.preset b/mkinitcpio.d/hook.preset
--- a/mkinitcpio.d/hook.preset	2023-03-19 08:24:59.000000000 -0600
+++ b/mkinitcpio.d/hook.preset	2023-03-23 10:27:58.544463061 -0600
@@ -1,17 +1,17 @@
-# mkinitcpio preset file for the '%PKGBASE%' package
+# mkinitcpio preset file for the '%KERNELBASE%' package
 
 ALL_config="/etc/mkinitcpio.conf"
-ALL_kver="/boot/vmlinuz-%PKGBASE%"
+ALL_kver="/boot/vmlinuz-%KERNELBASE%"
 ALL_microcode=(/boot/*-ucode.img)
 
 PRESETS=('default' 'fallback')
 
 #default_config="/etc/mkinitcpio.conf"
-default_image="/boot/initramfs-%PKGBASE%.img"
+default_image="/boot/initramfs-%KERNELBASE%.img"
 #default_uki="/efi/EFI/Linux/arch-%PKGBASE%.efi"
 #default_options="--splash /usr/share/systemd/bootctl/splash-arch.bmp"
 
 #fallback_config="/etc/mkinitcpio.conf"
-fallback_image="/boot/initramfs-%PKGBASE%-fallback.img"
+fallback_image="/boot/initramfs-%KERNELBASE%-fallback.img"
 #fallback_uki="/efi/EFI/Linux/arch-%PKGBASE%-fallback.efi"
 fallback_options="-S autodetect"
