diff --unified --recursive --text a/libalpm/scripts/mkinitcpio b/libalpm/scripts/mkinitcpio
--- a/libalpm/scripts/mkinitcpio	2022-11-20 13:18:56.000000000 -0700
+++ b/libalpm/scripts/mkinitcpio	2022-11-25 13:27:18.847308184 -0700
@@ -7,7 +7,7 @@
 
 process_preset() {
     if [[ -n "$pkgbase" && -e "$preset" ]]; then
-        if ! cmp "$preset" > /dev/null 2>&1 <(sed "s|%PKGBASE%|${pkgbase}|g" /usr/share/mkinitcpio/hook.preset); then
+        if ! cmp "$preset" > /dev/null 2>&1 <(sed "s|%KERNELBASE%|${kernelbase}|g" /usr/share/mkinitcpio/hook.preset); then
             if [[ ! -e "$preset.pacsave" ]]; then
                 # save the preset as pacsave
                 mv -- "$preset" "$preset.pacsave" && return 0
@@ -27,13 +27,13 @@
             mv -- "${preset}.pacsave" "$preset"
         else
             # create the preset from the template
-            sed "s|%PKGBASE%|${1}|g" /usr/share/mkinitcpio/hook.preset \
+            sed "s|%KERNELBASE%|${kernelbase}|g" /usr/share/mkinitcpio/hook.preset \
                 | install -Dm644 /dev/stdin "$preset"
         fi
     fi
 
     # always install the kernel
-    install -Dm644 "$line" "/boot/vmlinuz-${1}"
+    install -Dm644 "${line}" "/boot/vmlinuz-${kernelbase}"
 
     # compound args for each kernel
     args+=(-p "$1")
@@ -41,10 +41,10 @@
 
 remove_kernel() {
     # remove the actual kernel and images for the package being removed
-    kernel="/boot/vmlinuz-${1}"
-    preset="/etc/mkinitcpio.d/${1}.preset"
-    initramfs="/boot/initramfs-${1}.img"
-    fallback_initramfs="/boot/initramfs-${1}-fallback.img"
+    kernel="/boot/vmlinuz-${kernelbase}"
+    preset="/etc/mkinitcpio.d/${kernelbase}.preset"
+    initramfs="/boot/initramfs-${kernelbase}.img"
+    fallback_initramfs="/boot/initramfs-${kernelbase}-fallback.img"
     # remove the installed kernel
     rm -f -- "$kernel"
 
@@ -66,6 +66,11 @@
         continue
     fi
 
+    if ! read -r kernelbase > /dev/null 2>&1 < "${line%/vmlinuz}/kernelbase"; then
+        # this kernel has no kernelbase, use pkgbase
+        kernelbase="${pkgbase}"
+    fi
+
     case "$1" in
         install) install_kernel "$pkgbase";;
         remove) remove_kernel "$pkgbase";;
diff --unified --recursive --text a/mkinitcpio.d/example.preset b/mkinitcpio.d/example.preset
--- a/mkinitcpio.d/example.preset	2022-11-20 13:18:56.000000000 -0700
+++ b/mkinitcpio.d/example.preset	2022-11-25 13:25:08.956417074 -0700
@@ -21,8 +21,8 @@
 default_efi_image="/efi/EFI/Linux/arch-linux.efi"
 default_options=""
 
-#fallback_kver="3.0-ARCH"
+#fallback_kver="5.3.8-3-MANJARO"
 #fallback_config="/etc/mkinitcpio.conf"
-fallback_image="/tmp/initramfs-linux-fallback.img"
-fallback_efi_image="/efi/EFI/Linux/arch-linux-fallback.efi"
+fallback_image="/boot/initramfs-5.3-x86_64-fallback.img"
+fallback_efi_image="/efi/EFI/Manjaro/grubx64-fallback.efi"
 fallback_options="-S autodetect"
diff --unified --recursive --text a/mkinitcpio.d/hook.preset b/mkinitcpio.d/hook.preset
--- a/mkinitcpio.d/hook.preset	2022-11-20 13:18:56.000000000 -0700
+++ b/mkinitcpio.d/hook.preset	2022-11-25 13:26:02.945395342 -0700
@@ -1,14 +1,14 @@
 # mkinitcpio preset file for the '%PKGBASE%' package
 
 ALL_config="/etc/mkinitcpio.conf"
-ALL_kver="/boot/vmlinuz-%PKGBASE%"
+ALL_kver="/boot/vmlinuz-%KERNELBASE%"
 
 PRESETS=('default' 'fallback')
 
 #default_config="/etc/mkinitcpio.conf"
-default_image="/boot/initramfs-%PKGBASE%.img"
+default_image="/boot/initramfs-%KERNELBASE%.img"
 #default_options=""
 
 #fallback_config="/etc/mkinitcpio.conf"
-fallback_image="/boot/initramfs-%PKGBASE%-fallback.img"
+fallback_image="/boot/initramfs-%KERNELBASE%-fallback.img"
 fallback_options="-S autodetect"
