diff --git a/lib/util-iso.sh b/lib/util-iso.sh
index fedfb9d..f65d6ef 100644
--- a/lib/util-iso.sh
+++ b/lib/util-iso.sh
@@ -331,6 +331,7 @@ make_image_root() {
             echo 'MHWD64_IS_LIB32="'${multilib}'"' > "${path}/etc/mhwd-x86_64.conf"
         fi
 
+	cp "${tmp_dir}/custom-pacman.conf" "${path}/etc/pacman.conf" && sync
         pacman -Qr "${path}" > "${path}/rootfs-pkgs.txt"
         copy_overlay "${profile_dir}/root-overlay" "${path}"
 
@@ -353,11 +354,10 @@ make_image_desktop() {
         mount_fs_root "${path}"
 
         chroot_create "${path}" "${packages}"
-
+	cp "${tmp_dir}/custom-pacman.conf" "${path}/etc/pacman.conf" && sync
         pacman -Qr "${path}" > "${path}/desktopfs-pkgs.txt"
         cp "${path}/desktopfs-pkgs.txt" ${iso_dir}/$(gen_iso_fn)-pkgs.txt
         [[ -e ${profile_dir}/desktop-overlay ]] && copy_overlay "${profile_dir}/desktop-overlay" "${path}"
-
         if [[ -e "${path}/usr/share/calamares/branding/manjaro/calamares-sidebar.qml" ]]; then
             configure_branding "${path}"
             msg "Done [Distribution: Release ${dist_release} Codename ${dist_codename}]"
@@ -397,6 +397,7 @@ make_image_live() {
         mount_fs_select "${path}"
 
         chroot_create "${path}" "${packages}"
+	cp "${tmp_dir}/custom-pacman.conf" "${path}/etc/pacman.conf" && sync
 
         pacman -Qr "${path}" > "${path}/livefs-pkgs.txt"
         copy_overlay "${profile_dir}/live-overlay" "${path}"
@@ -429,6 +430,7 @@ make_image_mhwd() {
         mkdir -p ${path}${mhwd_repo}
 
         mount_fs_select "${path}"
+	cp "${tmp_dir}/custom-pacman.conf" "${path}/etc/pacman.conf" && sync
 
         reset_pac_conf "${path}"
 
@@ -612,7 +614,14 @@ get_pacman_conf(){
         info "detected: %s" "user-repos.conf"
         check_user_repos_conf "${user_conf}"
         conf=${tmp_dir}/custom-pacman.conf
-        cat ${DATADIR}/pacman-$pac_arch.conf ${user_conf} > "$conf"
+        a=$(($(grep -n "\[core\]" ${DATADIR}/pacman-$pac_arch.conf |sed -e 's/:.*//g') - 1))
+        b=$(wc -l ${DATADIR}/pacman-$pac_arch.conf |sed -e 's/ .*//g')
+        head -$a ${DATADIR}/pacman-$pac_arch.conf > "$conf"
+        cat ${user_conf} >> "$conf"
+        tail -$b ${DATADIR}/pacman-$pac_arch.conf >> "$conf"
+        if [[ "$profile" == "mabox" ]]; then
+            sed 's/manjaro-keyring/manjaro-keyring mabox-keyring/' -i "$conf"
+        fi
     else
         conf="${DATADIR}/pacman-$pac_arch.conf"
     fi
