diff --git a/lib/util-iso.sh b/lib/util-iso.sh
index 3931e55..7f3ae9c 100644
--- a/lib/util-iso.sh
+++ b/lib/util-iso.sh
@@ -331,6 +331,7 @@ make_image_root() {
             echo 'MHWD64_IS_LIB32="'${multilib}'"' > "${path}/etc/mhwd-x86_64.conf"
         fi
 
+	cp "${tmp_dir}/custom-pacman.conf" "${path}/etc/pacman.conf" && sync
         pacman -Qr "${path}" > "${path}/rootfs-pkgs.txt"
         copy_overlay "${profile_dir}/root-overlay" "${path}"
 
@@ -338,6 +339,7 @@ make_image_root() {
 
         configure_lsb "${path}"
 
+        cp "${DATADIR}/pacman-multilib.conf" "${path}/etc/pacman.conf" && sync
         clean_up_image "${path}"
         : > ${work_dir}/build.${FUNCNAME}
         msg "Done [Base installation] (rootfs)"
@@ -353,6 +355,7 @@ make_image_desktop() {
         mount_fs_root "${path}"
 
         chroot_create "${path}" "${packages}"
+	cp "${tmp_dir}/custom-pacman.conf" "${path}/etc/pacman.conf" && sync
 
         pacman -Qr "${path}" > "${path}/desktopfs-pkgs.txt"
         cp "${path}/desktopfs-pkgs.txt" ${iso_dir}/$(gen_iso_fn)-pkgs.txt
@@ -363,6 +366,7 @@ make_image_desktop() {
             msg "Done [Distribution: Release ${dist_release} Codename ${dist_codename}]"
         fi
 
+        cp "${DATADIR}/pacman-multilib.conf" "${path}/etc/pacman.conf" && sync
         reset_pac_conf "${path}"
 
         seed_snaps ${path}
@@ -394,6 +398,7 @@ make_image_live() {
         mount_fs_select "${path}"
 
         chroot_create "${path}" "${packages}"
+	cp "${tmp_dir}/custom-pacman.conf" "${path}/etc/pacman.conf" && sync
 
         pacman -Qr "${path}" > "${path}/livefs-pkgs.txt"
         copy_overlay "${profile_dir}/live-overlay" "${path}"
@@ -406,6 +411,7 @@ make_image_live() {
         
         configure_polkit_user_rules "${path}"
 
+        cp "${DATADIR}/pacman-multilib.conf" "${path}/etc/pacman.conf" && sync
         reset_pac_conf "${path}"
 
         umount_fs
@@ -423,6 +429,7 @@ make_image_mhwd() {
         mkdir -p ${path}${mhwd_repo}
 
         mount_fs_select "${path}"
+	cp "${tmp_dir}/custom-pacman.conf" "${path}/etc/pacman.conf" && sync
 
         reset_pac_conf "${path}"
 
@@ -436,6 +443,7 @@ make_image_mhwd() {
         cp ${DATADIR}/pacman-mhwd.conf ${path}/opt/mhwd
         make_repo "${path}"
         configure_mhwd_drivers "${path}"
+	cp "${DATADIR}/pacman-multilib.conf" "${path}/etc/pacman.conf" && sync
 
         umount_fs
         clean_up_image "${path}"
@@ -606,7 +614,7 @@ get_pacman_conf(){
         info "detected: %s" "user-repos.conf"
         check_user_repos_conf "${user_conf}"
         conf=${tmp_dir}/custom-pacman.conf
-        cat ${DATADIR}/pacman-$pac_arch.conf ${user_conf} > "$conf"
+        cat ${user_conf} ${DATADIR}/pacman-$pac_arch.conf > "$conf"
     else
         conf="${DATADIR}/pacman-$pac_arch.conf"
     fi
