diff --git a/lib/util-iso-image.sh b/lib/util-iso-image.sh
index 124fab4..144c548 100644
--- a/lib/util-iso-image.sh
+++ b/lib/util-iso-image.sh
@@ -369,6 +369,7 @@ copy_from_cache(){
     sed -i "s/.*\///" "$list"
 
     msg2 "Copying mhwd package cache ..."
+    sync
     rsync -v --files-from="$list" /var/cache/pacman/pkg "$1${mhwd_repo}"
 }
 
diff --git a/lib/util-publish.sh b/lib/util-publish.sh
index b1a4c43..6de972a 100644
--- a/lib/util-publish.sh
+++ b/lib/util-publish.sh
@@ -86,6 +86,7 @@ sync_dir(){
     msg "Start upload [%s] to [%s] ..." "$1" "${project}"
 
     while [[ $count -le $max_count ]]; do
+        sync
         rsync ${rsync_args[*]} --exclude '.latest*' --exclude 'index.html' --exclude 'links.txt' ${src_dir}/ ${server}/${target_dir}/
         if [[ $? != 0 ]]; then
             count=$(($count + 1))
diff --git a/lib/util-iso.sh b/lib/util-iso.sh
index edf50a0..8154067 100644
--- a/lib/util-iso.sh
+++ b/lib/util-iso.sh
@@ -323,6 +323,7 @@ make_image_root() {
         msg "Prepare [Base installation] (rootfs)"
         local path="${work_dir}/rootfs"
         mkdir -p ${path}
+        sync
 
         chroot_create "${path}" "${packages}" || die
 
@@ -349,6 +350,7 @@ make_image_desktop() {
         msg "Prepare [Desktop installation] (desktopfs)"
         local path="${work_dir}/desktopfs"
         mkdir -p ${path}
+        sync
 
         mount_fs_root "${path}"
 
@@ -393,6 +395,7 @@ make_image_live() {
         msg "Prepare [Live installation] (livefs)"
         local path="${work_dir}/livefs"
         mkdir -p ${path}
+        sync
 
         mount_fs_select "${path}"
 
@@ -434,6 +437,7 @@ make_image_mhwd() {
         msg "Prepare [drivers repository] (mhwdfs)"
         local path="${work_dir}/mhwdfs"
         mkdir -p ${path}${mhwd_repo}
+        sync
 
         mount_fs_select "${path}"
 
@@ -463,6 +467,7 @@ make_image_boot() {
         local boot="${iso_root}/boot"
 
         mkdir -p ${boot}
+        sync
 
         cp ${work_dir}/rootfs/boot/vmlinuz* ${boot}/vmlinuz-${target_arch}
 
