diff --git a/initcpio/install/miso_pxe_common b/initcpio/install/miso_pxe_common
index eec9a7e..3d935f1 100644
--- a/initcpio/install/miso_pxe_common
+++ b/initcpio/install/miso_pxe_common
@@ -8,9 +8,9 @@ build() {
     add_binary /usr/lib/initcpio/ipconfig /bin/ipconfig
 
     # Add hosts support files+dns
-    add_symlink /usr/lib/libnss_files.so.2 $(readlink /usr/lib/libnss_files.so.2)
+    #add_symlink /usr/lib/libnss_files.so.2 $(readlink /usr/lib/libnss_files.so.2)
     add_binary $(readlink -f /usr/lib/libnss_files.so.2)
-    add_symlink /usr/lib/libnss_dns.so.2 $(readlink /usr/lib/libnss_dns.so.2)
+    #add_symlink /usr/lib/libnss_dns.so.2 $(readlink /usr/lib/libnss_dns.so.2)
     add_binary $(readlink -f /usr/lib/libnss_dns.so.2)
 
     add_dir /etc
